hooks/checkout

```diff
@@ -203,6 +203,9 @@ main() {
 
   initialize_local_repo
 
+  # set origin in case the repo was changed by the pre-checkout hook
+  git remote set-url origin "${BUILDKITE_REPO}"
```

Comment:
This might break the checkout for any steps that don't use the checkout plugin. Can we save the old value of the origin and reset it back in an exit hook? Do exit hooks still run when the job is cancelled? Does it not matter because we want all the steps on our agents to use the checkout plugin anyway?

Reply:
yeah I think we should use the plugin in all the steps running on the same agents.